editor/utils/shortcuts.ts

```diff
@@ -0,0 +1,42 @@
+import { useCallback, useLayoutEffect, useRef, useEffect } from "react";
+
+/**
+ * Custom hook for adding keyboard shortcuts.
+ * 
+ * The shift key must be used in conjunction with the array of keys passed
+ * and the focused element can't be an input or text area
+ * in order to fire the handleKeyPress callback.
+ */
+export const useKeyboardShortcut = (
+  // Array of keys that should trigger the callback when used alongside the shift key
+  shortcutKeys: string[],
+  // Callback that handles what happens when the keys are pressed
+  callback: (event: KeyboardEvent) => void
+) => {
+  const callbackRef = useRef(callback);
+  useLayoutEffect(() => {
+    callbackRef.current = callback;
+  });
+  const handleKeyPress = useCallback(
+    (event: KeyboardEvent) => {
+      // Type guard to access tagName property, see https://iifx.dev/en/articles/156175355
+      if (event.target instanceof Element) {
+        // Don't trigger callback if focused element is an input or textarea
+        if (event.target.tagName === "INPUT" || event.target.tagName === "TEXTAREA") {
+          return
+        }
+        // Check if key in shortcutKeys array and used in conjunction with shift key
+        if (shortcutKeys.some((key) => event.shiftKey === true && event.key.toUpperCase() === key.toUpperCase())) {
```

Comment:
for future use cases elsewhere in app, we may want to do shortcut key combos that dont include the shiftKey in which case we could pass `shouldUseShift` or something as a param to this hook, but until then i think this is fine

Reply:
agreed!